### PR TITLE
Create an M3 issue template

### PR DESCRIPTION
**Pull request by richardartoul**
Ultimately we'd like to automate some of this stuff via an endpoint in the services themselves, but would be good to have a template to start with.

### CODE REVIEW

File: issue_template.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+### CPU / Heap Profiles
+
+CPU and heap profiles increase our ability to debug performance issues. All our services run with the [net/http/pprof](https://golang.org/pkg/net/http/pprof/) server enabled by default.
+
+Instructions for obtaining CPU / heap profiles for various services are below.
+
+#### M3Coordinator
+
+CPU
+`go tool pprof <HOST_NAME>:<PORT(default 7201)>/debug/pprof/profile?seconds=5`
```
Comment (prateek) [inline issue_template.md]: wire up curl versions of this and write out a file. (don't know if `go tool pprof -o` is a thing)

File: issue_template.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+3. What is the approximate series cardinality that the series is handling in a given time window? I.E How many unique time series are being measured?
+4. What is the hardware configuration (number CPU cores, amount of RAM, disk size and types, etc) that the service is running on? Is the service the only process running on the host or is it colocated with other software?
+5. What is the configuration of the service? Please include any YAML files, as well as namespace / placement configuration (with any sensitive information anonymized if necessary).
+6. How are you using the service? For example, are you performing read/writes to the service via Prometheus, or are you using a custom script?
+
+In addition to the above information, CPU and heap profiles are always greatly appreciated.
+
+### CPU / Heap Profiles
+
+CPU and heap profiles increase our ability to debug performance issues. All our services run with the [net/http/pprof](https://golang.org/pkg/net/http/pprof/) server enabled by default.
```
Comment (prateek) [inline issue_template.md]: nit: `s/increase our ability to/are critical to helping us`. 

further, add a line explicitly asking people to attach these profiles (it's implied right now)

File: issue_template.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Performance issues
+
+If the issue is performance related, please provide the following information along with a description of the issue that you're experiencing:
+
+1. What service is experiencing the performance issue? (M3Coordinator, M3DB, M3Aggregator, etc)
+2. Approximately how many datapoints per second is the service handling?
+3. What is the approximate series cardinality that the series is handling in a given time window? I.E How many unique time series are being measured?
+4. What is the hardware configuration (number CPU cores, amount of RAM, disk size and types, etc) that the service is running on? Is the service the only process running on the host or is it colocated with other software?
+5. What is the configuration of the service? Please include any YAML files, as well as namespace / placement configuration (with any sensitive information anonymized if necessary).
+6. How are you using the service? For example, are you performing read/writes to the service via Prometheus, or are you using a custom script?
```
Comment (prateek) [inline issue_template.md]: grafana dashboard

File: issue_template.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Filing M3 Issues
+
+## General Issues
+
+Please provide the following information along with a description of the issue that you're experiencing:
+
+1. What service is experiencing the issue? (M3Coordinator, M3DB, M3Aggregator, etc)
+2. What is the configuration of the service? Please include any YAML files, as well as namespace / placement configuration (with any sensitive information anonymized if necessary).
+3. How are you using the service? For example, are you performing read/writes to the service via Prometheus, or are you using a custom script?
```
Comment (prateek) [inline issue_template.md]: ask them to include repro instructions (if possible) too

File: issue_template.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Filing M3 Issues
+
+## General Issues
```
Comment (prateek) [inline issue_template.md]: describe what counts as a general issue v perf issue.

**Comment by richardartoul** [comment]
@justinjc Thats what this is, if you give a file the right name Github will use it

File: issue_template.md (hunk 1 above)
Comment (cw9) [reply issue_template.md]: it by default outputs the file locally

Comment (richardartoul) [reply issue_template.md]: it puts it in a weird directory though, changed them to CURLs so people dont need the go toolchain installed

**Pull request merged**